finesse/gui/interferometer_monitor.py

```diff
@@ -0,0 +1,204 @@
+"""Panel and widgets related to monitoring the interferometer."""
+from copy import deepcopy
+from typing import Optional
+
+from PySide6.QtCore import Qt
+from PySide6.QtWidgets import QGridLayout, QGroupBox, QLabel, QLineEdit
+
+from .led_icons import LEDIcon
+
+
+def get_vals_from_server() -> dict:
+    """Placeholder function for retrieving interferometer properties.
+
+    Returns:
+        data_table: A dictionary containing the physical properties being monitored
+    """
+    data_table = {
+        "PSF27 Temp": [28.151062, "deg. C"],
+        "Cryo Temp": [0.0, "deg. K"],
+        "Blackbody Hum": [2.463968, "%"],
+        "Source Temp": [70.007156, "deg. C"],
+        "Aux Volt": [6.285875, "V"],
+        "Aux Curr": [0.910230, "A"],
+        "Laser Curr": [0.583892, "A"],
+    }
+    return data_table
+
+
+class EM27Monitor(QGroupBox):
+    """Widgets to view the EM27 properties."""
+
+    def __init__(self, prop_labels: list[str], prop_units: list[Optional[str]]) -> None:
```

Comment:
I don't think we want the `prop_labels` and `prop_units` arguments here. What you've done is similar to what we've done for other placeholder controls, but I would request these values from the server, rather than passing them into the constructor (i.e. just use your `get_vals_from_server()`).

Reply:
I disagree here. If initialising this class depends on a connection to a server, it is necessary to mock that in order to test it. For this type of things, I would keep `__init__` as Dan has done it and implement a factory method (like the ones used for `LEDicon`) where  `get_vals_from_server()` is called.

Reply:
Maybe this should be done in a separate setter? My point was that if we add the controls dynamically, then in theory we won't know what `prop_labels` we will have upfront and will have to wait until we first poll the server before we know how many controls to add and what their labels will be etc.